src/auth/auth-config.ts

```diff
@@ -0,0 +1,576 @@
+export interface SAMLConfigServerResponse {
+  // Used when getting config.
+  // projects/${projectId}/inboundSamlConfigs/${providerId}
+  name?: string;
+  idpConfig?: {
+    idpEntityId?: string;
+    ssoUrl?: string;
+    idpCertificates?: Array<{
+      x509Certificate: string;
+    }>;
+    signRequest?: boolean;
+  };
+  spConfig?: {
+    spEntityId?: string;
+    callbackUri?: string;
+  };
+  displayName?: string;
+  enabled?: boolean;
+}
+
+/** The server side OIDC configuration request interface. */
+export interface OIDCConfigServerRequest {
+  clientId?: string;
+  issuer?: string;
+  displayName?: string;
+  enabled?: boolean;
+  [key: string]: any;
+}
+
+/** The server side OIDC configuration response interface. */
+export interface OIDCConfigServerResponse {
+  // Used when getting config.
+  // projects/${projectId}/oauthIdpConfigs/${providerId}
+  name?: string;
+  clientId?: string;
+  issuer?: string;
+  displayName?: string;
+  enabled?: boolean;
+}
+
+/** The public API response interface for listing provider configs. */
+export interface ListProviderConfigResults {
+  providerConfigs: AuthProviderConfig[];
+  pageToken?: string;
+}
+
+/** The public API request interface for updating a SAML Auth provider. */
+export interface SAMLUpdateAuthProviderRequest {
+  idpEntityId?: string;
+  ssoURL?: string;
+  x509Certificates?: string[];
+  rpEntityId?: string;
+  callbackURL?: string;
+  enableRequestSigning?: boolean;
+  enabled?: boolean;
+  displayName?: string;
+}
+
+/** The generic request interface for updating/creating a SAML Auth provider. */
+export interface SAMLAuthProviderRequest extends SAMLUpdateAuthProviderRequest {
+  providerId?: string;
+}
+
+/** The public API request interface for updating an OIDC Auth provider. */
+export interface OIDCUpdateAuthProviderRequest {
+  clientId?: string;
+  issuer?: string;
+  enabled?: boolean;
+  displayName?: string;
+}
+
+/** The generic request interface for updating/creating an OIDC Auth provider. */
+export interface OIDCAuthProviderRequest extends OIDCUpdateAuthProviderRequest {
+  providerId?: string;
+}
+
+/** The public API request interface for updating a generic Auth provider. */
+export type UpdateAuthProviderRequest = SAMLUpdateAuthProviderRequest | OIDCUpdateAuthProviderRequest;
+
+
+/**
+ * Defines the SAMLConfig class used to convert a client side configuration to its
+ * server side representation.
+ */
+export class SAMLConfig implements SAMLAuthProviderConfig {
+  public readonly enabled: boolean;
+  public readonly displayName?: string;
+  public readonly providerId: string;
+  public readonly idpEntityId: string;
+  public readonly ssoURL: string;
+  public readonly x509Certificates: string[];
+  public readonly rpEntityId: string;
+  public readonly callbackURL?: string;
+  public readonly enableRequestSigning?: boolean;
+
+  /**
+   * Static method convert a client side request to a SAMLConfigServerRequest.
```

Comment:
Just `Converts a client side request to a SAMLConfigServerRequest, which is the format accepted by the backend server.`.